src/ghga_service_commons/utils/multinode_storage.py

```diff
@@ -0,0 +1,53 @@
+# Copyright 2021 - 2023 Universität Tübingen, DKFZ, EMBL, and Universität zu Köln
+# for the German Human Genome-Phenome Archive (GHGA)
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+"""Configuration for multiple object storage nodes"""
+
+from hexkit.providers.s3 import S3Config, S3ObjectStorage
+from pydantic_settings import BaseSettings
+
+
+class ObjectStorageNodeConfig(BaseSettings):
+    """Configuration for one specific object storage node"""
+
+    bucket: str
```

Comment:
What if we need multiple buckets in the same service that are present on the same object storage. E.g. if need to move something from one bucket to another then it has to happen with the exact same credentials and the same ObjectStorage instance of hexkit.

Reply:
One option I can see is to turn the bucket into a dictionary. However, the Problem stays that it is difficult to generalize because it is very specific to the service which buckets do exist. Validation logic for check for the specification of the required buckets would thus need to live outside of this class.

Reply:
Not sure if I get your central problem point here...
The problem point here are services that are responsible for more than one bucket?

Reply:
Yes, but as clarified, there are no services that currently control more than one bucket. So this comment is not relevant anymore.